segmentation/include/pcl/segmentation/impl/crf_segmentation.hpp

```diff
@@ -71,7 +71,7 @@ pcl::CrfSegmentation<PointT>::~CrfSegmentation ()
 template <typename PointT> void
 pcl::CrfSegmentation<PointT>::setInputCloud (typename pcl::PointCloud<PointT>::Ptr input_cloud)
 {
-  if (input_cloud_ != NULL)
+  if (input_cloud_ != nullptr)
```

Comment:
Maybe just get rid of this? There is absolutely no reason to explicitly reset shared pointer before assignment,  `input_cloud_ = input_cloud;` is sufficient.

Reply:
There are other occurrences of similar needless reset in this PR, could you please also remove them?